cvat/apps/lambda_manager/views.py

```diff
@@ -0,0 +1,386 @@
+            attributes = {db_label.id:
+                {db_attr.name: db_attr.id for db_attr in db_label.attributespec_set.all()}
+                for db_label in db_labels}
+            labels = {db_label.name:db_label.id for db_label in db_labels}
+
+            # TODO: need to check 'cancel' operation
+            job = rq.get_current_job()
+            job.meta["progress"] = frame / db_task.data.size
+            job.save_meta()
+
+
+            # TODO: need to make "tags" and "tracks" are optional
+            # FIXME: need to provide the correct version here
+            results = {"version": 0, "tags": [], "shapes": [], "tracks": []}
+            for anno in annotations:
+                label_id = labels.get(anno["label"])
+                if label_id is not None:
+                    results["shapes"].append({
+                        "frame": frame,
+                        "label_id": label_id,
+                        "type": anno["type"],
+                        "occluded": False,
+                        "points": anno["points"],
+                        "z_order": 0,
+                        "group": None,
+                        "attributes": []
+                    })
+
+            serializer = LabeledDataSerializer(data=results)
+            if serializer.is_valid(raise_exception=True):
+                patch_task_data(db_task.id, results, "create")
+
+
+def return_response(success_code=status.HTTP_200_OK):
+    def wrap_response(func):
+        @wraps(func)
+        def func_wrapper(*args, **kwargs):
+            data = None
+            status_code = success_code
+            try:
+                data = func(*args, **kwargs)
+            except requests.ConnectionError as err:
+                status_code = status.HTTP_503_SERVICE_UNAVAILABLE
+                data = str(err)
+            except requests.HTTPError as err:
+                status_code = err.response.status_code
+                data = str(err)
+            except requests.Timeout as err:
+                status_code = status.HTTP_504_GATEWAY_TIMEOUT
+                data = str(err)
+            except requests.RequestException as err:
+                status_code = status.HTTP_500_INTERNAL_SERVER_ERROR
+                data = str(err)
+            except ValidationError as err:
+                status_code = err.code
+                data = err.message
+
+            return Response(data=data, status=status_code)
+
+        return func_wrapper
+    return wrap_response
+
+# TODO: need to protect the class. Only a registered use can get information
+# about available serverless functions.
+class FunctionViewSet(viewsets.ViewSet):
+    lookup_value_regex = '[a-zA-Z0-9_.-]+'
+    lookup_field = 'id'
+
+    @return_response()
+    def list(self, request):
+        gateway = LambdaGateway()
+        return [f.to_dict() for f in gateway.list()]
+
+    @return_response()
+    def retrieve(self, request, id):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/cvat/cvat/pullRequest?prid=5729179)